diff --git a/Makefile b/Makefile
index 22b6a98..ddb2657 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS = -Wall -Wextra -Werror -O2
+CFLAGS = -no-pie -Wall -Wextra -O2
 
 .PHONY: run
 run: kvm-hello-world
@@ -8,10 +8,10 @@ run: kvm-hello-world
 	./kvm-hello-world -l
 
 kvm-hello-world: kvm-hello-world.o payload.o
-	$(CC) $^ -o $@
+	$(CC) -no-pie $^ -o $@
 
 payload.o: payload.ld guest16.o guest32.img.o guest64.img.o
-	$(LD) -T $< -o $@
+	$(LD) -r -T $< -o $@
 
 guest64.o: guest.c
 	$(CC) $(CFLAGS) -m64 -ffreestanding -fno-pic -c -o $@ $^
